libraries/AP_GPS/AP_GPS_UBLOX.cpp

```diff
@@ -851,6 +866,59 @@ AP_GPS_UBLOX::_parse_gps(void)
         state.horizontal_accuracy = 0;
 #endif
         break;
+
+    case MSG_NAV_RELPOSNED: {
+        Debug("MSG_NAV_RELPOSNED RTK status Received");
+
+        state.rtk_age_ms    = 0xFFFFFFFF;
+
+    	state.rtk_baseline_x_mm = _buffer.rtk.rel_pos_n_cm*10+_buffer.rtk.rel_pos_hp_n_mm;
```

Comment:
This should be checking the `relPosValid` flag before copying any data. If we timeout with no real data for more then X seconds we should clear out the struct and unset the supports_mavlink_gps_rtk_message stuff.